jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/actions/PurgeQueueAction.kt

```diff
@@ -24,42 +25,53 @@ import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.Result
 import software.aws.toolkits.telemetry.SqsTelemetry
 
-class PurgeQueueAction :
-    SingleResourceNodeAction<SqsQueueNode>(message("sqs.purge_queue")),
-    CoroutineScope by ApplicationThreadPoolScope("PurgeQueueAction "),
-    DumbAware {
-    override fun actionPerformed(selected: SqsQueueNode, e: AnActionEvent) {
-        val project = selected.nodeProject
-        val client: SqsClient = project.awsClient()
+class PurgeQueueAction(
+    private val project: Project,
+    private val client: SqsClient,
+    private val queue: Queue
+) : DumbAwareAction(message("sqs.purge_queue.action")), CoroutineScope by ApplicationThreadPoolScope("PurgeQueueAction") {
+    override fun actionPerformed(e: AnActionEvent) {
+        val response = Messages.showOkCancelDialog(
+            project,
+            message("sqs.purge_queue.confirm", queue.queueName),
```

Comment:
I was actually going to suggest using [`com.intellij.openapi.ui.Messages#showConfirmationDialog`](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/openapi/ui/Messages.java#L387) - but it looks like it's deprecated in favour of `com.intellij.openapi.ui.MessageDialogBuilder`